tuist/tuist
# Align resource bundle accessor generation with SPM

Resolves https://github.com/tuist/tuist/issues/6149

### Short description 📝

Comply to SPM bundle accessors, both for internal and external module.
Following up on this [PR](https://github.com/tuist/tuist/pull/6124) with a more novel solution.

Currently Tuist have a mix of how it generates bundle accessors for both ObjC and Swift run time. This PR tries to unify them and comply to SPM logic.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

Use the `disableBundleAccessors: true` option.

In a target with Swift code and Resources, code will be generated and linked to the target to support the `Bundle.module` accessor - similar to SPM.

In a target with Objective-C code and Resources, code will be generated and linked to the target to support the `SWIFTPM_MODULE_BUNDLE` accessor macro - similar to SPM.

### Contributor checklist ✅


### Reviewer checklist ✅
